freqtrade/pairlist/PrecisionFilter.py

```diff
@@ -31,34 +41,34 @@ def _validate_precision_filter(self, ticker: dict, stoploss: float) -> bool:
         :param ticker: ticker dict as returned from ccxt.load_markets()
         :param stoploss: stoploss value as set in the configuration
                         (already cleaned to be 1 - stoploss)
-        :return: True if the pair can stay, false if it should be removed
+        :return: True if the pair can stay, False if it should be removed
         """
         stop_price = ticker['ask'] * stoploss
+
         # Adjust stop-prices to precision
         sp = self._exchange.price_to_precision(ticker["symbol"], stop_price)
+
         stop_gap_price = self._exchange.price_to_precision(ticker["symbol"], stop_price * 0.99)
         logger.debug(f"{ticker['symbol']} - {sp} : {stop_gap_price}")
+
         if sp <= stop_gap_price:
             self.log_on_refresh(logger.info,
                                 f"Removed {ticker['symbol']} from whitelist, "
                                 f"because stop price {sp} would be <= stop limit {stop_gap_price}")
             return False
+
         return True
 
     def filter_pairlist(self, pairlist: List[str], tickers: Dict) -> List[str]:
         """
         Filters and sorts pairlists and assigns and returns them again.
         """
-        stoploss = self._config.get('stoploss')
-        if stoploss is not None:
-            # Precalculate sanitized stoploss value to avoid recalculation for every pair
-            stoploss = 1 - abs(stoploss)
-        # Copy list since we're modifying this list
-        for p in deepcopy(pairlist):
-            ticker = tickers.get(p)
-            # Filter out assets which would not allow setting a stoploss
-            if not ticker or (stoploss and not self._validate_precision_filter(ticker, stoploss)):
-                pairlist.remove(p)
-                continue
+        if self._stoploss:
+            # Copy list since we're modifying this list
+            for p in deepcopy(pairlist):
+                ticker = tickers[p]
+                # Filter out assets which would not allow setting a stoploss
+                if not self._validate_precision_filter(ticker, self._stoploss):
+                    pairlist.remove(p)
```

Comment:
I am not convinced that this does the same thing than the old code.

pairlist may contain a pair not in tickers (assuming the database contains a pair no longer traded) ... which would now raise a KeyError - which is not handled here.

Now handling the above case (pair in pairlist not trade anymore) is a different topic - but it should not raise here anyway.

Reply:
Right, but when wrote it I assumed Filters are called after VolumePairList (remember our hot dicussion in Slack? 😉 ), which handles it.
But you are right here -- Filters can also follow StaticPairList, which does not deal with markets...

I'll add temporary this check into the Filters again, but I'll refactor it soon -- this cleanup should be done once, called from the pairlistmanager, instead of a check implemented in each of the Filters...

Reply:
yes i do remember - but that's the point of pairlists - you can use them in this sequence:
StaticPairlist - Filter1, Filter2, Volumepairlist (for sorting)

Reply:
How about this approach? (See the last commit: the pairlist is prepared only once, removing pairs with no ticker info if this is needed for filters)

TODO:
* couple of extra tests
* log removing pairs from the list with no ticker info available (??? is it needed?)

Reply:
Yeah, i would log it ... just so it's known that this pair was removed (only applies to static pairlist anyway ...)

Reply:
see the comment below on logging in pairlistmanager